src/main/java/core/basesyntax/dao/impl/FruitStorageDaoImpl.java

```diff
@@ -0,0 +1,31 @@
+package core.basesyntax.dao.impl;
+
+import core.basesyntax.dao.FruitStorageDao;
+import core.basesyntax.db.FruitStorage;
+import core.basesyntax.model.FruitTransaction;
+import java.util.HashMap;
+import java.util.Map;
+
+public class FruitStorageDaoImpl implements FruitStorageDao {
+    @Override
+    public void addFruit(FruitTransaction fruitTransaction) {
+        FruitStorage.fruitStorage.put(fruitTransaction.getFruit(), fruitTransaction.getQuantity());
+    }
+
+    @Override
+    public void increaseQuantity(FruitTransaction fruitTransaction) {
+        FruitStorage.fruitStorage.merge(fruitTransaction.getFruit(),
+                fruitTransaction.getQuantity(), Integer::sum);
+    }
+
+    @Override
+    public void decreaseQuantity(FruitTransaction fruitTransaction) {
+        FruitStorage.fruitStorage.merge(fruitTransaction.getFruit(),
+                -fruitTransaction.getQuantity(), Integer::sum);
+    }
+
+    @Override
+    public Map<String, Integer> getAllAsMap() {
+        return new HashMap<>(FruitStorage.fruitStorage);
```

Comment:
Returning a new `HashMap` instance in the `getAllAsMap` method is a good practice to avoid exposing the internal storage. However, ensure that this is the intended behavior as it will not reflect subsequent changes to the `fruitStorage` after this map is returned.